data/registers/adc_l111x.yaml

```diff
@@ -0,0 +1,1309 @@
+  - name: SOFTWARE
+    description: The interrupt or event line is in software mode. Software must clear the RIS.
+    value: 1
+  - name: HARDWARE
+    description: The interrupt or event line is in hardware mode. The hardware (another module) clears automatically the associated RIS flag.
+    value: 2
+enum/PWRDN:
+  bit_size: 1
+  variants:
+  - name: AUTO
+    description: ADC is powered down on completion of a conversion if there is no pending trigger.
+    value: 0
+  - name: MANUAL
+    description: ADC remains powered on as long as it is enabled through software.
+    value: 1
+enum/PWREN_KEY:
+  bit_size: 8
+  variants:
+  - name: KEY
+    value: 38
+enum/RES:
+  bit_size: 2
+  variants:
+  - name: BIT_12
+    description: 12-bits resolution.
+    value: 0
+  - name: BIT_10
+    description: 10-bits resolution.
+    value: 1
+  - name: BIT_8
+    description: 8-bits resolution.
+    value: 2
+enum/RESET_KEY:
+  bit_size: 8
+  variants:
+  - name: KEY
+    value: 177
+enum/SAMPCLK:
+  bit_size: 2
+  variants:
+  - name: ULPCLK
+    description: ULPCLK is the source of ADC sample clock.
+    value: 0
+  - name: SYSOSC
+    description: SYSOSC is the source of ADC sample clock.
+    value: 1
+  - name: HFCLK
+    description: 'HFCLK clock is the source of ADC sample clock. Note : HFCLK may not be available on all the devices.'
+    value: 2
+enum/SAMPCNT:
+  bit_size: 5
+  variants:
+  - name: MIN
+    description: Minimum value.
+    value: 0
+  - name: MAX
+    description: Maximum value.
+    value: 24
+enum/SAMPMODE:
+  bit_size: 1
+  variants:
+  - name: AUTO
+    description: Sample timer high phase is used as sample signal.
+    value: 0
+  - name: MANUAL
+    description: Software trigger is used as sample signal.
+    value: 1
+enum/SC:
+  bit_size: 1
+  variants:
+  - name: STOP
+    description: When SAMPMODE is set to MANUAL, clearing this bit will end the sample phase and the conversion phase will start. When SAMPMODE is set to AUTO, writing 0 has no effect.
+    value: 0
+  - name: START
+    description: When SAMPMODE is set to MANUAL, setting this bit will start the sample phase. Sample phase will last as long as this bit is set. When SAMPMODE is set to AUTO, setting this bit will trigger the timer based sample time.
+    value: 1
+enum/SCLKDIV:
+  bit_size: 3
+  variants:
+  - name: DIV_BY_1
+    description: Do not divide clock source.
+    value: 0
+  - name: DIV_BY_2
+    description: Divide clock source by 2.
+    value: 1
+  - name: DIV_BY_4
+    description: Divide clock source by 4.
+    value: 2
+  - name: DIV_BY_8
+    description: Divide clock source by 8.
+    value: 3
+  - name: DIV_BY_16
+    description: Divide clock source by 16.
+    value: 4
+  - name: DIV_BY_24
+    description: Divide clock source by 24.
+    value: 5
+  - name: DIV_BY_32
+    description: Divide clock source by 32.
+    value: 6
+  - name: DIV_BY_48
+    description: Divide clock source by 48.
+    value: 7
+enum/STARTADD:
```

Comment:
This enum can be deleted. This is just enum variants that directly map to an index.